tests/native-lib/fail/trace_read.rs

```diff
@@ -0,0 +1,26 @@
+//@only-target: linux
+//@only-target: gnu
+//@only-target: x86
+//@only-on-host
+//@compile-flags: -Zmiri-native-lib-enable-tracing
+
+extern "C" {
+    fn do_nothing();
+}
+
+fn main() {
+    unexposed_reachable_alloc();
+}
+
+fn unexposed_reachable_alloc() {
+    let inner = 42;
+    let intermediate = &raw const inner;
+    let exposed = (&raw const intermediate).expose_provenance();
+    unsafe { do_nothing() };
+    let invalid: *const i32 = std::ptr::with_exposed_provenance(intermediate.addr());
+    let valid: *const *const i32 = std::ptr::with_exposed_provenance(exposed);
```

Comment:
What's the point of `valid`?

Reply:
Expanded this so that the native read actually happens and we properly check that it does expose whatever it reads to but no more